researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/TaskHelper.java

```diff
@@ -189,19 +188,25 @@ void uploadTaskResult(TaskResult taskResult, Archive.WithBridgeConfig withBridge
             }
         }
 
+        // bridgeManagerProvider.getApplicationContext().getFilesDir() + File.separator +
         String archiveFilename = taskId + "_" + UUID.randomUUID().toString() + ".zip";
 
         bridgeManagerProvider.getUploadManager()
-                .upload(archiveFilename, bridgeArchiveBuilder.build()).toCompletable().await();
+                .queueUpload(archiveFilename, bridgeArchiveBuilder.build())
+                .doOnSuccess(uploadFile -> {
+                    logger.debug("Calling uploadToS3() to upload pending uploads");
+
+                    // At this point, the upload request has been processed and saved,
+                    // so it is safe to delete the temporary data logger files
+                    for (Result result : results) {
+                        if (result instanceof FileResult) {
+                            FileResult fileResult = (FileResult) result;
+                            DataLoggerManager.getInstance().deleteFileStatus(fileResult.getFile());
```

Comment:
I'm not entirely sure if deleteFileStatus is thread-safe.  I didn't intend on it to be called off the  main thread.